ang/test/civicase/activity/feed/directives/activity-month-nav.directive.spec.js

```diff
@@ -2,6 +2,66 @@
 
 (function (_) {
   describe('civicaseActivityMonthNav', function () {
+    describe('Activity Month Nav Directive', function () {
+      var $compile, $rootScope, $scope, $timeout, heightSpy;
+
+      beforeEach(module('civicase', 'civicase.templates'));
+
+      beforeEach(inject(function (_$compile_, _$rootScope_, _$timeout_) {
+        $compile = _$compile_;
+        $rootScope = _$rootScope_;
+        $timeout = _$timeout_;
+
+        $scope = $rootScope.$new();
+
+        heightSpy = spyOn(CRM.$.fn, 'height');
+      }));
+
+      describe('on init', function () {
+        var heightSpyCallSelectors, expectedHeightSpyCallSelectors;
+
+        beforeEach(function () {
+          initDirective(true);
+          $timeout.flush();
+
+          heightSpyCallSelectors = _.map(heightSpy.calls.all(), function (callObj) {
+            return callObj.object.selector;
+          });
+          expectedHeightSpyCallSelectors = [
+            '#toolbar',
+            '.civicase__case-body_tab',
+            '.civicase__activity-month-nav',
+            '.civicase__dashboard__tab-container ul.nav',
+            '.civicase__activity-filter'
+          ];
+        });
+
+        it('sets the height of the month nav', function () {
+          expect(_.union(heightSpyCallSelectors, expectedHeightSpyCallSelectors))
+            .toEqual(expectedHeightSpyCallSelectors);
```

Comment:
I think this method to test the height of the element is not reliable. I could call the selectors without storing their individual heights and the test would still pass.

I think the best way to test this is to add a height style for each element:

```js
var html = `<div class="civicase__activity-month-nav" style="height: 100px"></div>
<div id="toolbar" style="height: 100px"></div>
...
```

Then test that the `height` spy was called with `calc(100vh - 500px)`.

Reply:
I tried to do the same before but given up I could not figure out, so I let it go. But after this comment I looked at it again, and figured that I had spied `height` function and was wondering why the height was not getting set properly.
Fixed it thanks.